### PR TITLE
Fix environment variables in Exec plugin

### LINKED ISSUE #60: ENOENT Error with AWS EKS
I am trying to use kubebox with AWS EKS. I updated to 0.6.0 which supports the exec auth. I have been unsuccessful in connecting due to the following error:

spawn aws-iam-authenticator ENOENT 

Pasting in a token doesn't seem to work either.

Any ideas? What am I missing?

### PR DESCRIPTION
**Pull request by johnpoth**
fixes #60

### CODE REVIEW

File: lib/auth/exec.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 const { execFile } = require('child_process'),
+      proc         = require('process'),
```
Comment (astefanutti) [inline lib/auth/exec.js]: I would avoid requiring `process` and maybe rename the eponym variable to `exec` and just use `process.env`.

File: lib/auth/exec.js [hunk 2]
```diff
@@ -34,9 +35,13 @@ class ExecAuth {
       promise = Promise.resolve(this.auth_provider);
     } else {
       promise = new Promise((resolve, reject) => {
-        const env = {};
+        let env;
```
Comment (astefanutti) [inline lib/auth/exec.js]: Shouldn't it default to `process.env`, irrespectively of `this.auth_provider.env`?

File: lib/auth/exec.js (hunk 1 above)
Comment (johnpoth) [reply lib/auth/exec.js]: Done ( renamed it environment  )

Comment (astefanutti) [reply lib/auth/exec.js]: I meant just use `process.env` inline everywhere as `process` is global, but I realise my first comment was not crystal clear 😉.

File: lib/auth/exec.js (hunk 2 above)
Comment (johnpoth) [reply lib/auth/exec.js]: I think you would then modify the process.env Object by reference when adding values to env?

That shouldn't be a problem unless you were switching between contexts where environment variables would all end up in process.env and produce undesired behavior...

Comment (astefanutti) [reply lib/auth/exec.js]: Here I meant that env should be the concatenation of `process.env` and `this.auth_provider.env`. Hope that's clearer 😀.

Comment (astefanutti) [reply lib/auth/exec.js]: > I think you would then modify the process.env Object by reference when adding values to env?

You can do `let env = Object.assign({}, process.env, ...)`

File: lib/auth/exec.js (hunk 1 above)
Comment (johnpoth) [reply lib/auth/exec.js]: Haha much clearer :+1:

File: lib/auth/exec.js (hunk 2 above)
Comment (johnpoth) [reply lib/auth/exec.js]: Done

Comment (johnpoth) [reply lib/auth/exec.js]: I just remembered why I didn't do it that way in the first place: we are now always copying process.env even when we don't need to

Comment (astefanutti) [reply lib/auth/exec.js]: I think that's acceptable. I prefer conciseness over performance in that case.

**Pull request merged**